xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/CommentsTab.java

```diff
@@ -55,11 +56,27 @@ public String getCurrentAuthor()
 
     public boolean isCommentFormShown()
     {
+        toggleComments();
         WebElement commentForm =
             getDriver().findElement(By.xpath("//form[@id='AddComment']/fieldset[@id='commentform']"));
         return commentForm.isDisplayed();
     }
 
+    private void toggleComments()
+    {
+        String commentFormId = "AddComment";
+        String openFormId = "openCommentForm";
+        XWikiWebDriver driver = getDriver();
+        // is the comments has not already been toggled.
+        if (!driver.findElementById(commentFormId).isDisplayed()) {
+            // we wait for the form button to be loaded.
+            driver.waitUntilElementIsVisible(By.id(openFormId));
```

Comment:
Why do you need to wait on this button? Should be loaded with all comments no? So when the tab is loaded this button should be loaded too.

Reply:
The comment area is loaded asynchronously. We need to wait for the button to be visible (ie, the comments to be loaded) before acting on it.

Reply:
From what I see the way to obtain a CommentsTab is by calling `ViewPage#openCommentsDocExtraPane` which apparently wait for the content to be loaded. I guess we could improve that a bit by putting this wait in the CommentsTab constructor. But not sure it should wait there.

Reply:
I think `ViewPage#openCommentsDocExtraPane` waits for the tab to switch to the comments. But it does not wait for the actual content of the comments, loaded asynchronously, is actually displayed.
An alternative to the current code is to wait for `#commentscontent` to be visible in `ViewPage#openCommentsDocExtraPane`, that way we dont have to wait here.

Reply:
btw @manuelleduc would have been good to use this as a pretext for refactoring CommentAsAdminTest and CommentAsGuestTest for docker tests.